app/models/concerns/nameable.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+module Nameable
+  extend ActiveSupport::Concern
+
+  def generate_github_repo_name
+    @suffix_number = 0
+
+    loop do
+      break unless GitHubRepository.present?(organization.github_client, full_name)
+      @suffix_number += 1
+    end
+
+    suffixed_repo_name
+  end
+
+  private
+
+  def base_name
+    @base_name ||= "#{slug}-#{name}"
+  end
+
+  def full_name
+    "#{organization_login}/#{suffixed_repo_name}"
+  end
+
+  def suffixed_repo_name
+    return base_name if @suffix_number.zero?
+
+    suffix = "-#{@suffix_number}"
+    base_name.truncate(100 - suffix.length) + suffix
```

Comment:
@tarebyte Yeah, but if we use `suffix` as the omission here. The suffix will only be added when the string is truncated, which means that the suffix will never show up if the string length is less than 100.